### PR TITLE
memdump: add shellcode detect

### PR DESCRIPTION
**Pull request by skvl**
A message like this should appear (with `-o kv` option):
```
memdump Time=1617703930.430326,PID=3036,PPID=2992,TID=3040,ProcessName="\Device\HarddiskVolume2\Users\John\Desktop\sample.exe",Method=NtFreeVirtualMemory,DumpReason="Possible shellcode detected",DumpPID=3036,DumpAddr=C0000,DumpSize=0x1000,DumpFilename="c0000_7265fc3339a44c62",SN=19
```

### CODE REVIEW

**Comment by tklengyel** [comment]
@drakvuf-jenkins This is OK to test

File: src/plugins/memdump/memdump.cpp [hunk 1]
```diff
@@ -537,6 +537,116 @@ static event_response_t terminate_process_hook_cb(drakvuf_t drakvuf, drakvuf_tra
...
+    }
+    if (temp_pid == *ctx->pid)
+    {
+        PRINT_DEBUG("[MEMDUMP] Found remote process base! Getting dtb..\n");
+        drakvuf_get_process_dtb(drakvuf, eprocess, ctx->dtb);
+        *ctx->process = eprocess;
+    }
+}
+
+static bool get_process_by_handle(drakvuf_t drakvuf, drakvuf_trap_info_t* info, uint64_t handle, addr_t* process, addr_t* dtb)
```
Comment (tklengyel) [inline src/plugins/memdump/memdump.cpp:558]: Looks like this function may be useful at other places too, so move it to `libdrakvuf`?

Comment (skvl) [reply src/plugins/memdump/memdump.cpp:558]: I guess so. I will do it as soon as possible.

Comment (skvl) [reply src/plugins/memdump/memdump.cpp]: Done.

**Comment by skvl** [comment]
@tklengyel what is the error with Windows 10 test?

**Comment by tklengyel** [comment]
@drakvuf-jenkins Retest this please

**Comment by tklengyel** [comment]
@drakvuf-jenkins This is OK to test

**Comment by skvl** [comment]
@tklengyel, I have re-based the branch

**Comment by tklengyel** [comment]
@drakvuf-jenkins Retest this please

**Comment by tklengyel** [comment]
@drakvuf-jenkins Retest this please

**Comment by tklengyel** [comment]
@drakvuf-jenkins Retest this please

**Pull request merged**